### PR TITLE
Pytorch ghost trainer

### PR DESCRIPTION
**Pull request by andrewcoh**
### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/policy/torch_policy.py [hunk 1]
```diff
@@ -249,13 +249,13 @@ def increment_step(self, n_steps):
         return self.get_current_step()
 
     def load_weights(self, values: List[np.ndarray]) -> None:
-        pass
+        self.actor_critic.load_state_dict(values)
 
     def init_load_weights(self) -> None:
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/policy/torch_policy.py:254]: We dont need this function

File: ml-agents/mlagents/trainers/policy/tf_policy.py [hunk 2]
```diff
@@ -152,6 +152,8 @@ def create_tf_graph(self) -> None:
         # We do an initialize to make the Policy usable out of the box. If an optimizer is needed,
         # it will re-load the full graph
         self.initialize()
+        # Create assignment ops for Ghost Trainer
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/policy/tf_policy.py:155]: A bit strange to have Ghost specific code in the TF policy. Isn't there another reason to call init_load_weights here ?

File: ml-agents/mlagents/trainers/ppo/trainer.py [hunk 3]
```diff
@@ -217,7 +217,10 @@ def _update_policy(self):
...
 
     def create_tf_policy(
-        self, parsed_behavior_id: BehaviorIdentifiers, behavior_spec: BehaviorSpec
+        self,
+        parsed_behavior_id: BehaviorIdentifiers,
+        behavior_spec: BehaviorSpec,
+        create_graph: bool = False,
     ) -> TFPolicy:
         """
         Creates a PPO policy to trainers list of policies.
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/ppo/trainer.py:226]: Not sure what create_graph does. Can you add a comment?

File: ml-agents/mlagents/trainers/policy/torch_policy.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/policy/torch_policy.py:254]: Maybe add a comment and mark for removal when TF is deprecated

File: ml-agents/mlagents/trainers/policy/tf_policy.py (hunk 2 above)
Comment (andrewcoh) [reply ml-agents/mlagents/trainers/policy/tf_policy.py:155]: Previously, this was tf specific code in the ghost....

I don't need to call it ghost specific but the ghost trainer is the only thing that uses the load/get methods

File: ml-agents/mlagents/trainers/ppo/trainer.py (hunk 3 above)
Comment (andrewcoh) [reply ml-agents/mlagents/trainers/ppo/trainer.py:226]: added. It sets the `create_tf_graph` flag in the `TFPolicy` constructor

File: ml-agents/mlagents/trainers/policy/tf_policy.py (hunk 2 above)
Comment (vincentpierre) [reply ml-agents/mlagents/trainers/policy/tf_policy.py:155]: Ok, I think we can do without that comment but I don't mind leaving it either.

**Pull request merged**